### PR TITLE
Feature/spline python agent 8 configuration

### LINKED ISSUE #8: Flexible agent configuration
There has to be a way to configure the agent using a config file or environment variables, so that the same code executed in different environments would behave differently.
What should be configured:
- lineage dispatcher and its properties
- agent mode (enabled/disabled)

### PR DESCRIPTION
**Pull request by wajda**
closes #8 

Adds a flexible configuration framework for the agent

### CODE REVIEW

**Review by TerezaHrubesova — APPROVED** [review]

**Review by TerezaHrubesova — APPROVED** [review]

**Review by AB006J9 — APPROVED** [review]

File: src/spline_agent/context.py [hunk 1]
```diff
@@ -18,66 +18,66 @@
...
-        if self._name is not None:
+        if self.__name is not None:
             logger.warning(f"Tracking context property 'name' is reassigned: "
-                           f"old value '{self._name}', new value '{value}'")
-        self._name = value
+                           f"old value '{self.__name}', new value '{value}'")
+        self.__name = value
 
     @property
     def inputs(self) -> tuple[DataSource, ...]:
```
Comment (lsulak) [inline src/spline_agent/context.py:48]: You are mixing an old typing style with the new one. Not a big deal though.

`tuple` (the standard type, ) vs `Tuple` (from typing module)

* https://peps.python.org/pep-0585/
* https://stackoverflow.com/questions/39458193/using-list-tuple-etc-from-typing-vs-directly-referring-type-as-list-tuple-etc

so if you are confident that you are always gonna be using Python 3.9 or higher, just use ones with lowercase and remove the imports from typing module

File: src/spline_agent/context.py [hunk 2]
```diff
@@ -18,66 +18,66 @@
...
-        return tuple(self._ins)
```
Comment (lsulak) [inline src/spline_agent/context.py:49]: `__inst` is a list, `inputs` is a tuple, which is a bit inconsistent. Perhaps there is a good reason for it, but you are basically representing 1 thing with two different data structures

File: src/spline_agent/exceptions.py [hunk 3]
```diff
@@ -12,16 +12,29 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 
-class LineageTrackingContextNotInitialized(BaseException):
-    """ Lineage harvesting context was not properly initialized"""
+class LineageTrackingContextNotInitializedError(BaseException):
```
Comment (lsulak) [inline src/spline_agent/exceptions.py:15]: deriving from BaseException is not best practice; it's even mentiond in the documentation: https://stackoverflow.com/questions/60745841/inheriting-from-baseexception-vs-exception#:~:text=Note%20that%20while%20the%20default,s%20thread%20equivalent%2C%20the%20Task.

Rather, derive from `Exception`, as explained in the SO post and/or the PEP8 guide: https://peps.python.org/pep-0008/#programming-recommendations

**Comment by lsulak** [comment]
I guess that, since this is an OSS, you will want to publish this into Pypi. It should be trivial using Poetry using poetry publish command (but first you will need to create a new Pypi account & register a new project there & prepare a new API token - https://pypi.org/help/#apitoken).

File: src/spline_agent/context.py (hunk 1 above)
Comment (wajda) [reply src/spline_agent/context.py:48]: fixed

File: src/spline_agent/context.py (hunk 2 above)
Comment (wajda) [reply src/spline_agent/context.py:49]: Both are different things. A readonly property `input` is a part of the class contract, while `__ins` is a private mutable member holding the state. The rational is encapsulation of the mutable state. `inputs` property doesn't return a pointer to the `__ins` as a different type. Instead, it returns a new instance of type tuple that effectively is a immutable shallow copy of the `__ins`.

File: src/spline_agent/exceptions.py (hunk 3 above)
Comment (wajda) [reply src/spline_agent/exceptions.py:15]: Nice one! My Java intuition suggested me that it's the `Exception` that must be root of the hierarchy with the `BaseException` being one of its subclasses. It turned out to be vice versa in Python 😲 

Fixed.

**Pull request merged**